qutebrowser/browser/history.py

```diff
@@ -68,23 +68,28 @@ class WebHistory(QWebHistoryInterface):
         _old_miss: How many times an URL was not found in _old_urls.
     """
 
-    changed = pyqtSignal()
+    item_added = pyqtSignal(HistoryEntry)
 
     def __init__(self, parent=None):
         super().__init__(parent)
         self._lineparser = lineparser.AppendLineParser(
             standarddir.data(), 'history', parent=self)
-        self._old_urls = set()
+        self._old_urls = {}
         with self._lineparser.open():
             for line in self._lineparser:
-                _time, url = line.rstrip().split(maxsplit=1)
-                self._old_urls.add(url)
+                atime, url = line.rstrip().split(maxsplit=1)
+                # This de-duplicates history entries. We keep later ones in the
+                # file which usually the last ones accessed. If you want
```

Comment:
*which usually the last ones accessed* doesn't make sense, but I also don't understand if it was *which usually **are** the last ones accessed* - do you mean "in the dict"?